### ISSUE TITLE
Wiremock's SpringHttpServer does not support FaultInjectors

### ISSUE THREAD

**Issue by gaqzi** [spring-cloud/spring-cloud-contract#270]
I have in other projects used the Wiremock [Fault Injectors] to test edge cases for failed requests, for example dump some random data and then close the connection before the request is "finished". 

But with the SpringHttpServer fault injectors are not available. It would be really nice if they were. :)

What I'm trying to do:

```java
        wireMockRule.stubFor(request
                                 .willReturn(aResponse()
                                                 .withFault(Fault.RANDOM_DATA_THEN_CLOSE)
                                 ));
```

The response I get back is `418` with a body of `No fault injector is configured!`.

[Fault Injectors]: http://wiremock.org/docs/simulating-faults/

**Comment by dsyer** [comment]
Do you have a full test case that you can share that would pass if this were supported?

**Comment by dsyer** [comment]
The Wiremock library only supports fault injection with Jetty in native mode (i.e. it doesn't apply for Spring Cloud Contract). We could add support for a `FaultInjectorFactory` to be configured, but the implementation would have to be custom, unless Wiremock can be changed to allow a generic fault injector for all servlet containers. If you have an idea how to implement that, please go ahead and try it.

**Comment by dsyer** [comment]
Actually, it looks like the Jetty version will work in a Spring Boot app. I'd love to have Tomcat and Undertow versions.

**Comment by dsyer** [comment]
Tomcat works with that change (sort of). Probably good enough to move forward.

**Comment by marcingrzejszczak** [comment]
Ok - I'll patch 1.0.x and merge it tomorrow

**Comment by dsyer** [comment]
Merged in d3f6ad9. Please open another ticket for Undertow.

**Comment by gaqzi** [comment]
@dsyer <3
